tests/Settings/EmbeddersTest.php

```diff
@@ -51,4 +52,57 @@ public function testResetEmbedders(): void
 
         self::assertSame(self::DEFAULT_EMBEDDER, $embedders);
     }
+
+    public function testHuggingFacePooling(): void
+    {
+        $embedder = [
+            'source' => 'huggingFace',
+            'model' => 'sentence-transformers/all-MiniLM-L6-v2',
+            'pooling' => 'useModel',
+        ];
+
+        $promise = $this->index->updateEmbedders([
+            'embedder_name' => [
+                'source' => 'huggingFace',
+                'model' => 'sentence-transformers/all-MiniLM-L6-v2',
+                'pooling' => 'useModel',
+            ],
+        ]);
+
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $embedders = $this->index->getEmbedders();
+
+        self::assertSame($embedder['source'], $embedders['embedder_name']['source']);
+        self::assertSame($embedder['model'], $embedders['embedder_name']['model']);
+        self::assertSame($embedder['pooling'], $embedders['embedder_name']['pooling']);
+    }
+
+    public function testCompositeEmbedder(): void
+    {
+        $http = new Client($this->host, getenv('MEILISEARCH_API_KEY'));
+        $http->patch('/experimental-features', ['compositeEmbedders' => true]);
+
+        $embedder = [
+            'source' => 'composite',
+            'searchEmbedder' => [
+                'source' => 'huggingFace',
+                'model' => 'sentence-transformers/all-MiniLM-L6-v2',
+            ],
+            'indexingEmbedder' => [
+                'source' => 'huggingFace',
+                'model' => 'sentence-transformers/all-MiniLM-L6-v2',
+            ],
+        ];
+
+        $promise = $this->index->updateEmbedders([
+            'embedder_name' => $embedder,
+        ]);
+
+        $this->assertIsValidPromise($promise);
+        $this->index->waitForTask($promise['taskUid']);
+
+        $embedders = $this->index->getEmbedders();
```

Comment:
what this does test?

Reply:
The test ensures the new "composite embedders" request format is valid. I assert this line 103:

```php
$this->assertIsValidPromise($promise);
```

Lines 104-106 were not useful. I removed them.